packages/devtools_app/lib/src/file_import.dart

```diff
@@ -204,6 +190,43 @@ class _FileImportContainerState extends State<FileImportContainer> {
   }
 }
 
+Future<void> importFileFromPicker({
+  @required List<String> acceptedTypes,
+  @required void Function(DevToolsJsonFile) importHandler,
+}) async {
+  final acceptedTypeGroups = [XTypeGroup(extensions: acceptedTypes)];
+  final file = await openFile(acceptedTypeGroups: acceptedTypeGroups);
+  final data = jsonDecode(await file.readAsString());
+  final lastModifiedTime = await file.lastModified();
+  final devToolsJsonFile = DevToolsJsonFile(
+    data: data,
+    name: file.name,
+    lastModifiedTime: lastModifiedTime,
+  );
+  importHandler(devToolsJsonFile);
```

Comment:
we be both async and have a callback? Simpler to just return the json from this method and then call the import handler after awaiting.

Reply:
done